x/delegation/keeper/delegation.go

```
@@ -131,26 +131,38 @@ func (k *Keeper) UndelegateFrom(ctx sdk.Context, params *delegationtype.Delegati
 	}
+	undelegationID := k.GetUndelegationID(ctx)
 	// record Undelegation event
 	r := delegationtype.UndelegationRecord{
 		StakerId:              stakerID,
 		AssetId:               assetID,
 		OperatorAddr:          params.OperatorAddress.String(),
 		TxHash:                params.TxHash.String(),
-		IsPending:             true,
-		LzTxNonce:             params.LzNonce,
+		UndelegationId:        undelegationID,
 		BlockNumber:           uint64(ctx.BlockHeight()),
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion int64 -> uint64

(gosec)